contrib/src/main/java/com/datatorrent/contrib/enrich/FixedWidthFSLoader.java

```diff
@@ -0,0 +1,248 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.enrich;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.classification.InterfaceStability;
+
+import com.google.common.collect.Maps;
+import com.google.common.primitives.Ints;
+import com.univocity.parsers.fixed.FixedWidthFieldLengths;
+import com.univocity.parsers.fixed.FixedWidthParser;
+import com.univocity.parsers.fixed.FixedWidthParserSettings;
+
+import com.datatorrent.contrib.parser.AbstractCsvParser.FIELD_TYPE;
+import com.datatorrent.contrib.parser.AbstractCsvParser.Field;
+
+/**
+ * This implementation of {@link FSLoader} is used to load data from fixed width
+ * file.User needs to set {@link FixedWidthFSLoader#fieldDescription} to specify
+ * field information.
+ */
+@InterfaceStability.Evolving
+public class FixedWidthFSLoader extends FSLoader
+{
+
+  private transient List<Field> fields;
+  /**
+   * Indicates whether first line of the file is a header. Default is false
+   */
+  private transient boolean hasHeader;
+
+  /**
+   * Specifies information related to fields in fixed-width file. Format is
+   * [NAME]:[FIELD_TYPE]:[WIDTH]:[date format if FIELD_TYPE is DATE] FIELD_TYPE
+   * can take on of the values of {@link FIELD_TYPE} i.e BOOLEAN, DOUBLE,
+   * INTEGER, FLOAT, LONG, SHORT, CHARACTER, STRING, DATE e.g.
+   * Year:INTEGER:4,Make:STRING:5,Model:STRING:40,Description:STRING:40,
+   * Price:DOUBLE:8,Date:DATE:10:\"dd:mm:yyyy\". Date format needs to be within
+   * quotes (" ")
+   */
+  @NotNull
+  private transient String fieldDescription;
+
+  /**
+   * Array containing headers
+   */
+  private transient String[] header;
+  /**
+   * Array containing width of each field
+   */
+  private transient int[] width;
+  /**
+   * Padding character. Default is white space.
+   */
+  private transient char padding = ' ';
+  private transient FixedWidthParser fixedWidthParser;
+  private transient boolean initialized;
+  /**
+   * Map holding the format for date fields
+   */
+  private transient Map<String, String> fieldDateFormat = Maps.newHashMap();
+
+  private static final Logger logger = LoggerFactory.getLogger(FixedWidthFSLoader.class);
+
+  public FixedWidthFSLoader()
```

Comment:
Default constructor is automatically inserted by JVM.. No need of empty constructor.